src/app/auth/services/auth.service.ts

```diff
@@ -0,0 +1,21 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { IUser } from '../models/auth/auth.model';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthService {
+
+  constructor(private http: HttpClient) { }
+  createUser(user:IUser)
+  {
+    this.http.post<IUser>('https://todo-api-nestjs.herokuapp.com/auth/register',user).subscribe(data=>{console.log('yes',data);
```

Comment:
1. move the `https://todo-api-nestjs.herokuapp.com/` to the environment file and use it here
2. don't subscribe in the service, return the Observable and subscribe in the component
3. save the returned token in the local storage
4. the return value is not `IUser` please check and adjust